querybook/server/lib/result_store/stores/file_store.py

```diff
@@ -27,25 +27,43 @@ def __init__(self, uri: str):
         self.uri = get_file_uri(uri)
 
     def start(self):
+        self._chunk_size=100000
         self._chunks_length = 0
+        self._chunks = []
         os.makedirs(self.uri_dir_path, exist_ok=True)
 
+    def _reset_variables(self):
+        self._chunks = []
+        self._chunks_length = 0
+
     def write(self, data: str):
-        # write each line into csv
+        # Append each line to a list and then dump at once
         data_len = len(data)
         if (
             QuerybookSettings.DB_MAX_UPLOAD_SIZE > 0
             and self._chunks_length + data_len > QuerybookSettings.DB_MAX_UPLOAD_SIZE
         ):
             return False
-
+        
         self._chunks_length += data_len
+        # with open(self.uri, "a") as result_file:
+        #     result_file.write(data)
+
+        if len(self._chunks) == self._chunk_size:
+            self.unload()
+            self._reset_variables()
+
+        self._chunks.append(data)
+        return True
+
+    def unload(self):
+        data = "".join(self._chunks)
         with open(self.uri, "a") as result_file:
             result_file.write(data)
-        return True
 
     def end(self):
-        pass
+        if len(self._chunks) < self._chunk_size:
```

Comment:
this checks seem unnecessary to me, as it should not happen

Reply:
This check would be used when the length of list appended (self._chunks) is less than the batch size defined as env variable. 
ie, if the number of rows retrieved from query is less than env variable or after making batches, the length of the last  batch would be less than the env variable.

This is tested and is working as expected.

Reply:
updated the same.